subsys/net/ip/udp.c

```diff
@@ -147,13 +147,23 @@ struct net_udp_hdr *net_udp_input(struct net_pkt *pkt,
 				  struct net_pkt_data_access *udp_access)
 {
 	struct net_udp_hdr *udp_hdr;
+	u16_t length;
 
 	udp_hdr = (struct net_udp_hdr *)net_pkt_get_data(pkt, udp_access);
 	if (!udp_hdr || net_pkt_set_data(pkt, udp_access)) {
 		NET_DBG("DROP: corrupted header");
 		goto drop;
 	}
 
+	length = ntohs(udp_hdr->len);
+	if (length < sizeof(struct net_udp_hdr) ||
```

Comment:
Could it be that the first check (length < ) is redundant? The second check verifies that the length is correct, hence it should work by itself?

Reply:
Indeed, moreover that net_pkt_get_data() can be successful if only payload length is at least of sizeof(struct net_udp_hdr). So checking if length is equal to payload length is indeed sufficient then.